src/hssm/hssm.py

```diff
@@ -403,19 +402,18 @@ def _transform_params(
                 param = _create_param(
                     param_dict, model_config, is_parent=name == self._parent
                 )
-                params.append(param)
+                params[name] = param
 
         for param_str in self.list_params:
             if param_str not in processed:
                 param = _create_param(
                     param_str, model_config, is_parent=param_str == self._parent
                 )
-                params.append(param)
+                params[param_str] = param
 
-        if len(params) != len(self.list_params):
-            raise ValueError("Please provide a correct set of priors")
+        sorted_params = {k: params[k] for k in self.list_params}
```

Comment:
'Ordered dict' might be useful here?

Reply:
Not sure if this kind of 'sorting' is actually valid for the dict structure in python (which I think operates as a set on the keys). So not sure if you are really guaranteeing the sort here.

Reply:
Yeah from Python 3.7 on the dictionaries are sorted in the order of insertion

Reply:
ah great. didn't actually know that :).